src/System.Diagnostics.TraceSource/tests/TraceClassTests.cs

```diff
@@ -1,122 +1,375 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System;
-using System.Diagnostics;
-using System.IO;
 using Xunit;
 
 namespace System.Diagnostics.TraceSourceTests
 {
-    internal static class HelperMethods
+    using Method = TestTraceListener.Method;
+
+    public class TraceClassTests
     {
-        public static string GetFileContents(string fileName)
+        public TraceClassTests()
         {
-            // Read the contents of the TextFile.
-            using (FileStream fs = File.Open(fileName, FileMode.Open))
-            {
-                byte[] buffer = new byte[fs.Length];
-                fs.Read(buffer, 0, (int)fs.Length);
-                return System.Text.Encoding.UTF8.GetString(buffer, 0, buffer.Length);
-            }
+            // restore Trace static state
+            Trace.IndentLevel = 0;
+            Trace.IndentSize = DiagnosticsConfiguration.IndentSize;
+            Trace.UseGlobalLock = DiagnosticsConfiguration.UseGlobalLock;
+            Trace.AutoFlush = DiagnosticsConfiguration.AutoFlush;
+            Trace.Listeners.Clear();
+            Trace.Listeners.Add(new DefaultTraceListener());
         }
 
-        public static bool CheckStrings(string fileName, string expectedString)
+        [Fact]
+        public void AutoFlushTest()
         {
-            string actualString = GetFileContents(fileName);
-            if (expectedString.Equals(actualString))
-            {
-                return true;
-            }
+            Trace.AutoFlush = false;
+            Assert.False(Trace.AutoFlush);
+            Trace.AutoFlush = true;
+            Assert.True(Trace.AutoFlush);
+        }
 
-            return false;
+        [Fact]
+        public void UseGlobalLockTest()
+        {
+            Trace.UseGlobalLock = true;
+            Assert.True(Trace.UseGlobalLock);
+            Trace.UseGlobalLock = false;
+            Assert.False(Trace.UseGlobalLock);
```

Comment:
Should each individual test try to restore the world to the state before the test, i.e. Trace.UseGlobalLock should be reset at the end to whatever it was at the beginning?